src/phoenix/server/api/mutations/project_mutations.py

```diff
@@ -32,11 +33,20 @@ async def delete_project(self, info: Info[Context, None], id: GlobalID) -> Query
         return Query()
 
     @strawberry.mutation(permission_classes=[IsAuthenticated])  # type: ignore
-    async def clear_project(self, info: Info[Context, None], id: GlobalID) -> Query:
-        project_id = from_global_id_with_expected_type(global_id=id, expected_type_name="Project")
+    async def clear_project(self, info: Info[Context, None], input: ClearProjectInput) -> Query:
+        project_id = from_global_id_with_expected_type(
+            global_id=input.id, expected_type_name="Project"
+        )
         delete_statement = delete(models.Trace).where(models.Trace.project_rowid == project_id)
+        if input.end_time is not None:
+            delete_statement = delete_statement.where(models.Trace.start_time < input.end_time)
         async with info.context.db() as session:
-            await session.execute(delete_statement)
+            try:
+                await session.execute(delete_statement)
+                await session.commit()
+            except Exception:
+                await session.rollback()
+                raise
             if cache := info.context.cache_for_dataloaders:
                 cache.invalidate(ClearProjectSpansEvent(project_rowid=project_id))
```

Comment:
rollback/commit is automatic. the erroneous portion is actually the placement of the invalidate statement
```suggestion
            await session.execute(delete_statement)
        if cache := info.context.cache_for_dataloaders:
            cache.invalidate(ClearProjectSpansEvent(project_rowid=project_id))
```